tests/vmi_configuration_test.go

```diff
@@ -1573,41 +1541,42 @@ var _ = Describe("Configurations", func() {
 				Expect(err).ToNot(HaveOccurred())
 				tests.WaitForSuccessfulVMIStart(cpuVmi)
 
+				output := tests.RunCommandOnVmiPod(cpuVmi, []string{"grep", "-m1", "model name", "/proc/cpuinfo"})
+
+				niceName := parseCPUNiceName(output)
+
 				By("Expecting the VirtualMachineInstance console")
 				Expect(libnet.WithIPv6(console.LoginToCirros)(cpuVmi)).To(Succeed())
 
 				By("Checking the CPU model under the guest OS")
 				console.SafeExpectBatch(cpuVmi, []expect.Batcher{
-					&expect.BSnd{S: fmt.Sprintf("grep '%s' /proc/cpuinfo\n", libvirtCpuModel)},
-					&expect.BExp{R: "model name"},
+					&expect.BSnd{S: fmt.Sprintf("grep '%s' /proc/cpuinfo\n", niceName)},
+					&expect.BExp{R: fmt.Sprintf(".*model name.*%s.*", niceName)},
 				}, 10)
 			})
 		})
 
 		Context("when CPU features defined", func() {
-			It("[test_id:3123]should start a Virtaul Machine with matching features", func() {
+			It("[test_id:3123]should start a Virtual Machine with matching features", func() {
+				supportedCPUFeatures := tests.GetSupportedCPUFeatures(nodes.Items[0])
+				Expect(len(supportedCPUFeatures)).ToNot(Equal(0))
 				cpuVmi.Spec.Domain.CPU = &v1.CPU{
 					Features: []v1.CPUFeature{
 						{
-							Name: cpuFeatures[0],
+							Name: supportedCPUFeatures[0],
 						},
 					},
 				}
 
+				fmt.Println("using " + supportedCPUFeatures[0] + " flag")
+
 				By("Starting a VirtualMachineInstance")
 				cpuVmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(cpuVmi)
 				Expect(err).ToNot(HaveOccurred())
 				tests.WaitForSuccessfulVMIStart(cpuVmi)
 
 				By("Expecting the VirtualMachineInstance console")
 				Expect(libnet.WithIPv6(console.LoginToCirros)(cpuVmi)).To(Succeed())
-
-				By("Checking the CPU features under the guest OS")
-				Expect(console.SafeExpectBatch(cpuVmi, []expect.Batcher{
-					&expect.BSnd{S: fmt.Sprintf("grep %s /proc/cpuinfo\n", cpuFeatures[0])},
-					&expect.BExp{R: "flags"},
-				}, 10)).To(Succeed())
```

Comment:
why do we not continue to introspect the guest here?

Reply:
Because libvirt reports some features which are not in /proc/cpuinfo and then this test will be flaky